calendarsubscription/calendarsubscription.go

```diff
@@ -40,3 +44,45 @@ func (cs CalendarSubscription) Normalize() (*CalendarSubscription, error) {
 
 	return &cs, nil
 }
+
+func (cs CalendarSubscription) RenderICalFromShifts(shifts []oncall.Shift, reminderMinutes []int) ([]byte, error) {
+	type iCalOptions struct {
+		Shifts          []oncall.Shift `json:"s,omitempty"`
+		ReminderMinutes []int          `json:"r, omitempty"`
+	}
+
+	var iCalTemplate = `
+		BEGIN:VCALENDAR
+		VERSION:2.0
+		PRODID:-//ZContent.net//Zap Calendar 1.0//EN
+		CALSCALE:GREGORIAN
+		METHOD:PUBLISH
+		{{range .Shifts}}
+		BEGIN:VEVENT
+		SUMMARY:On-Call
```

Comment:
Should we include the schedule name here, since users will be able to subscribe to multiple schedules? e.g. `On-call for GoAlert Primary Schedule`

Or perhaps we could set the calendar name under the `VCALENDAR` section (a `NAME` property is supported for the calendar title: https://icalendar.org/New-Properties-for-iCalendar-RFC-7986/5-1-name-property.html) and keep the `SUMMARY` simple for each `VEVENT` as something like `On-call shift`

Reply:
That sounds good! Would it also be worth including the calendar subscription name too?

Reply:
Let's wait until the final pass when we test everything together on actual devices to add names -- I don't want us to spin until we can actually try it.

Then we can try different things, see what is the best UX for diff. calendar providers.

Reply:
Sure! We can add more UX then after some testing, I'll revert my changes for the name field for now.